cloud/logutils.go

```
@@ -0,0 +1,79 @@
+package azurestackhci
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	mocerrors "github.com/microsoft/moc/pkg/errors"
+	"k8s.io/klog"
+)
+
+type MocResourceType string
+
+const (
+	LoadBalancer     MocResourceType = "LoadBalancer"
+	VipPool          MocResourceType = "VipPool"
+	VirtualNetwork   MocResourceType = "VirtualNetwork"
+	NetworkInterface MocResourceType = "NetworkInterface"
+	Disk             MocResourceType = "Disk"
+	VirtualMachine   MocResourceType = "VirtualMachine"
+	KeyVault         MocResourceType = "KeyVault"
+	Secret           MocResourceType = "Secret"
+	Group            MocResourceType = "Group"
+)
+
+type MocOperation string
+
+const (
+	CreateOrUpdate MocOperation = "CreateOrUpdate"
+	Delete         MocOperation = "Delete"
+	Get            MocOperation = "Get"
+)
+
+type OperationLog struct {
+	Timestamp      string      `json:"timestamp"`
+	ParentResource string      `json:"parent_resource"`
+	Resource       string      `json:"resource"`
+	FilterKeyword  string      `json:"filter_keyword"`
+	Action         string      `json:"action"`
+	Params         interface{} `json:"params"`
+	ErrorCode      string      `json:"error_code"`
+	Message        string      `json:"msg"`
+}
+
+func WriteMocOperationLog(operation MocOperation, crResourceName string, mocResourceType MocResourceType, mocResourceName string, params interface{}, err error) {
+	if (operation == Delete || operation == Get) && ResourceNotFound(err) {
+		err = nil
```

Are we sure we want to swallow this error?  
What if the error/bug is that we returned resource not found when we didn't expect it.
E.g. let's say MOC stack has a *bug* and createorupdate() returned resource not found.  Our telemetry would not reflect that error.
If we want our teleemtry to be effective, we should follow the calls inputs and outputs as is.
Any interpertation should happen in the queries.

---

thanks for suggestion. agree that better to return original error, and filter in kusto